main/src/hadd.cxx

```diff
@@ -24,7 +24,10 @@
   \param -ff  The compression level used is the one specified in the first input
 
   \param -k   Skip corrupt or non-existent files, do not exit
-  \param -O   Re-optimize basket size when merging TTree
+  \param -O   If hadd is used to merge TTree objects, compress the whole target
```

Comment:
This is 'only' a side effect of the main purpose of the option.   The main purpose is really to let the TTree decide anew what the basket size should be based on the current setup, etc...

Note #11245 is really a bug, so the 'tool' to change the compression is -f.

Reply:
Alright, I thought that option `-O` was intended also for making sure that all the branches in a TTree are properly recompressed.

I still think that the current description for the option is a bit too generic and unclear for an average user: "re-optimize basket size" doesn't tell on the basis of what the basket size is re-optimized. Also, basket is not a very user-friendly term imo.

Maybe a new wording could be
```
If hadd is used to merge TTree objects, let the target tree decide how to re-organize its internal [structure/layout].
In particular, it can change the number and size of the clusters (i.e. different groups of entries) in the tree. 
This option is most often used to improve a tree that has clusters with bad sizes (e.g. too large/small or too varied) or no clusters at all, which can lead to poor I/O performance.
```